api/service/syncing/syncing.go

```diff
@@ -761,14 +775,33 @@ Loop:
 			if size > SyncLoopBatchSize {
 				size = SyncLoopBatchSize
 			}
-			err := ss.ProcessStateSync(startHash[:], size, bc, worker)
+
+			retryCount := 0
+			var err error
+			for err == nil && retryCount < stateSyncRetryLimit {
```

Comment:
need to nil the error in order to retry, otherwise, the for loop check will just pass

Reply:
or maybe just remove "err == nil"